scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/CredentialValidator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.job.factories.spi.model.validator;
+
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.common.exception.PermissionException;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.ModelValidatorContext;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ValidationException;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 27/08/19
+ */
+public class CredentialValidator implements Validator<String> {
+    private static final Logger logger = Logger.getLogger(CredentialValidator.class);
+
+    @Override
+    public String validate(String parameterValue, ModelValidatorContext context) throws ValidationException {
+        if (context == null || context.getScheduler() == null) {
+            // Sometimes the workflow is submitted without scheduler instance (e.g., submitted from catalog).
```

Comment:
in case of the catalog, a workflow is not submitted, a workflow is just parsed and stored in the catalog. the logger warn is not necessary (you can put it as debug if you want) as it would pollute the catalog log file (basically, we know what we are doing).

Reply:
Done. Changed to debug.